adal/src/main/java/com/microsoft/aad/adal/ServerRespondingWithRetryableException.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.microsoft.aad.adal;
+
+/**
+ * server responding with retryable error.
+ */
+class ServerRespondingWithRetryableException extends AuthenticationException {
+    static final long serialVersionUID = 1;
+    
+    /**
+     * Constructs a new ServerRespondingWithRetryableException with message.
+     * @param code The error code for {@link ServerRespondingWithRetryableException}.
+     * @param msg Detailed message for the reason causing the exception.
+     */
+    public ServerRespondingWithRetryableException(ADALError code, String msg) {
```

Comment:
>ADALError code [](start = 50, length = 14)

I think the exception should have default error code, and the two constructor should only taken in errorMessage and throwable.